src/index.js

```diff
@@ -2,9 +2,89 @@
 // Do not delete or rename this file ********
 
 // An example of how you tell webpack to use a CSS (SCSS) file
+import User from './User';
+import Traveler from './Traveler';
+import TravelAgent from './TravelAgent';
+import domUpdates from '../src/domUpdates';
+import fetch from 'node-fetch';
 import './css/base.scss';
 
 // An example of how you tell webpack to use an image (also need to link to it in the index.html)
 import './images/turing-logo.png'
 
-console.log('This is the JavaScript entry file - your code begins here.');
+
+let user;
+let travelAgent;
+let traveler;
+let users;
+let destinations;
+let trips;
+let loginButton = document.querySelector('.login-button');
+
+loginButton.addEventListener('click', () => {
+    attemptLogin()
+});
+
+Promise.all([
+    fetch('https://fe-apps.herokuapp.com/api/v1/travel-tracker/data/travelers/travelers/')
+        .then(response => response.json()),
+    fetch('https://fe-apps.herokuapp.com/api/v1/travel-tracker/data/trips/trips')
+        .then(response => response.json()),
+    fetch('https://fe-apps.herokuapp.com/api/v1/travel-tracker/data/destinations/destinations')
+        .then(response => response.json())
+])
+    .then(data => createDatasets(data[0], data[1], data[2]))
+    .catch(err => console.error(err))
+
+const createDatasets = (userData, destinationData, tripsData) => {
+    users = userData.travelers;    
+    destinations = destinationData.destinations;    
+    trips = tripsData.trips;    
+}
+
+const attemptLogin = () => {
+    let usernameInput = document.getElementById('username-input').value;
+    let passwordInput = document.getElementById('password-input').value;
+    let newID = retrieveUserID(usernameInput);
+    if (newID) {
+        createUser(newID);
+        user.logIn(passwordInput);
+        checkLogInStatus();
+    }
+}
+
+const retrieveUserID = (username) => {
+    if (username === 'agency') {
+        return 'agency';
+    } else if (username.includes('traveler')) {
+        return parseInt(username.substr(8));
+    } else {
+        return undefined;
+    }
+}
+
+const createUser = (newID) => {
+    if (newID === 'agency') {
+        user = new User(newID);
+        travelAgent = new TravelAgent(newID);
+        return travelAgent;
+    } else {
+        let userInfo = finduserByID(newID);
+        user = new User(userInfo);
+        traveler = new Traveler(newID, userInfo);
+        return traveler;
+    }
+}
+
+const checkLogInStatus = () => {
```

Comment:
what exactly is the logged in status for? if you instantiate a particular user based in id, does it need it?

Reply:
The check log in status determines whether the user has entered the correct password and shows the dashboard if they have. I'm not sure if it needs it, I might rethink some of my class structure later to simplify it.